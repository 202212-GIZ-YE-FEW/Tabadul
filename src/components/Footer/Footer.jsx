import React from "react";

import {
    Container,
    Footercss,
    Imgsocial,
    Imgsociallogo,
    PagesList,
    Section,
    Social,
    SocialLink,
    LinkSoial,
} from "./Footer.style";

function Footer() {
    return (
        <Footercss>
            <Container>
                <Section>
                    <Imgsociallogo
                        src='/images/footerlogo.svg'
                        alt=''
                        width={50}
                        height={50}
                    />
                </Section>

                <Section>
                    <PagesList href='/Aboutus'>About us</PagesList>
                    <PagesList href='/Blogs'>Blogs</PagesList>
                    <PagesList href='/Products'>Products</PagesList>
                </Section>

                <Section>
                    <p>Contact Us</p>

                    <Social>
                        <LinkSoial href='#'>
                            <SocialLink>
                                <Imgsocial
                                    src='/images/facebook .svg'
                                    alt=''
                                    width={50}
                                    height={50}
                                />
                            </SocialLink>
                        </LinkSoial>
                        <LinkSoial href='#'>
                            <SocialLink>
                                <Imgsocial
                                    src='/images/instagram (2).svg'
                                    alt=''
                                    width={50}
                                    height={50}
                                />
                            </SocialLink>
                        </LinkSoial>
                        <LinkSoial href='#'>
                            <SocialLink>
                                <Imgsocial
                                    src='/images/linkedin (2).svg'
                                    alt=''
                                    width={50}
                                    height={50}
                                />
                            </SocialLink>
                        </LinkSoial>
                        <LinkSoial href='#'>
                            <SocialLink>
                                <Imgsocial
                                    src='/images/share.svg'
                                    alt=''
                                    width={50}
                                    height={50}
                                />
                            </SocialLink>
                        </LinkSoial>
                    </Social>
                </Section>

                <Section>
                    <p>Address</p>
                    <p>street 01,07</p>
                    <p>Yemen, Sanaa</p>
                </Section>
            </Container>
        </Footercss>
    );
}

export default Footer;
