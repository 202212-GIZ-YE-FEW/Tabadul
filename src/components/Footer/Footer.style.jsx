import styled from "@emotion/styled";
import Image from "next/image";

export const Footercss = styled.div`
    width: 100%;
    background-color: #33956d;
    color: #fff;
    @media screen and (max-width: 700px) {
        position: relative;
        bottom: 0px;
    }
`;

export const Container = styled.div`
    display: flex;
    padding: 3rem;

    @media screen and (max-width: 700px) {
        flex-wrap: wrap;
        flex-direction: column;
        padding: 0.5rem;
    }
`;

export const Section = styled.div`
    flex-basis: 50%;
    padding: 4px 20px;

    @media screen and (max-width: 700px) {
        margin: 5px 0;
    }
`;
export const PagesList = styled.a`
    display: block;
    text-decoration: none;
    color: white;
    margin-top: 10px;
    line-height: 2;
`;

export const Social = styled.div`
    margin: 20px 0;
`;

export const SocialLink = styled.a`
    padding: 3px;
    direction: none;
`;

export const Imgsocial = styled(Image)`
    width: auto;
    height: 32px;
    padding: 5px;
`;

export const Imgsociallogo = styled(Image)`
    width: 100px;
    height: 100px;
    margin-top: 30px;
`;
