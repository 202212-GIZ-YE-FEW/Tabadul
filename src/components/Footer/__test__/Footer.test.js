import renderer from "react-test-renderer";

import Footer from "../Footer";

jest.mock("react-i18next", () => ({
    // this mock makes sure any components using the translate hook can use it without a warning being shown
    useTranslation: () => {
        return {
            t: (str) => str,
            i18n: {
                changeLanguage: () => new Promise(() => {}),
            },
        };
    },
}));

jest.mock("next/router", () => ({
    useRouter() {
        return {
            pathname: "",
            // ... whatever else you you call on `router`
        };
    },
}));

it("renders correctly", () => {
    const tree = renderer.create(<Footer />).toJSON();
    expect(tree).toMatchSnapshot();
});
