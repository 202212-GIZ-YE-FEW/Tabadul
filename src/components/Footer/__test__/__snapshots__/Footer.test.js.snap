// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="css-16e8e6z"
>
  <div
    className="css-kp9m8d"
  >
    <div
      className="css-xvkl1c"
    >
      <img
        alt=""
        className="css-n2b87z"
        data-nimg="1"
        decoding="async"
        height={50}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/images/footerlogo.svg"
        style={
          Object {
            "color": "transparent",
          }
        }
        width={50}
      />
    </div>
    <div
      className="css-xvkl1c"
    >
      <a
        className="css-1dbkwr8"
        href="/Aboutus"
      >
        About us
      </a>
      <a
        className="css-1dbkwr8"
        href="/Blogs"
      >
        Blogs
      </a>
      <a
        className="css-1dbkwr8"
        href="/Products"
      >
        Products
      </a>
    </div>
    <div
      className="css-xvkl1c"
    >
      <p>
        Contact Us
      </p>
      <div
        className="css-18icqaw"
      >
        <a
          className="css-8xe971"
          href="#"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          <a
            className="css-1i6zk58"
          >
            <img
              alt=""
              className="css-3g96gi"
              data-nimg="1"
              decoding="async"
              height={50}
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src="/images/facebook .svg"
              style={
                Object {
                  "color": "transparent",
                }
              }
              width={50}
            />
          </a>
        </a>
        <a
          className="css-8xe971"
          href="#"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          <a
            className="css-1i6zk58"
          >
            <img
              alt=""
              className="css-3g96gi"
              data-nimg="1"
              decoding="async"
              height={50}
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src="/images/instagram (2).svg"
              style={
                Object {
                  "color": "transparent",
                }
              }
              width={50}
            />
          </a>
        </a>
        <a
          className="css-8xe971"
          href="#"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          <a
            className="css-1i6zk58"
          >
            <img
              alt=""
              className="css-3g96gi"
              data-nimg="1"
              decoding="async"
              height={50}
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src="/images/linkedin (2).svg"
              style={
                Object {
                  "color": "transparent",
                }
              }
              width={50}
            />
          </a>
        </a>
        <a
          className="css-8xe971"
          href="#"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          <a
            className="css-1i6zk58"
          >
            <img
              alt=""
              className="css-3g96gi"
              data-nimg="1"
              decoding="async"
              height={50}
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src="/images/share.svg"
              style={
                Object {
                  "color": "transparent",
                }
              }
              width={50}
            />
          </a>
        </a>
      </div>
    </div>
    <div
      className="css-xvkl1c"
    >
      <p>
        Address
      </p>
      <p>
        street 01,07
      </p>
      <p>
        Yemen, Sanaa
      </p>
    </div>
  </div>
</div>
`;
